src/exception/property-exceptions/repo/entity.ts

```diff
@@ -97,7 +97,8 @@ const entityExceptionRepository = {
 				upsert: true,
 				new: true,
 				returnDocument: 'after',
-			}).lean(true);
+			}).lean<EntityException>(true);
+			if (!doc) throw new Error();
```

Comment:
Source of errors which was not caught before
Better Types w/ library upgrades also make this necessary
`doc` is potentially `null` in which case we throw an error

Reply:
Can you expand a bit on the errors that this was throwing please? I want to ensure no further investigation needed. `doc` should never be null because we're upserting with findOneAndUpdate. DB level should be throwing correctly and caught correctly.